storage/innobase/log/log0recv.cc

```diff
@@ -2846,7 +2846,8 @@ recv_sys_t::parse_mtr_result recv_sys_t::parse(source &l, bool if_exists)
                                    last_offset)
                 : file_name_t::initial_flags;
               if (it == recv_spaces.end())
-                ut_ad(!file_checkpoint || space_id == TRX_SYS_SPACE ||
```

Comment:
I checked how this assert is prevented when we are storing the records and a little above we have the following check precisely taking care of the case where the space might have been deleted later.

```
2721     else if (store && file_checkpoint && !is_predefined_tablespace(space_id))
2722     {
2723       recv_spaces_t::iterator i= recv_spaces.lower_bound(space_id);
2724       if (i != recv_spaces.end() && i->first == space_id);
2725       else if (lsn < file_checkpoint)
2726         /* We have not seen all records between the checkpoint and
2727         FILE_CHECKPOINT. There should be a FILE_DELETE for this
2728         tablespace later. */
2729         recv_spaces.emplace_hint(i, space_id, file_name_t("", false));
```
Since for store=false, we do go thorough all the records and fill recv_spaces based on FILE_ records, it might be better to include this case also for `store=false` and emplace the dummy record.  We can then consistently use the assert for all cases.